buku

```diff
@@ -584,48 +585,60 @@ class BukuDb:
 
         return (conn, cur)
 
-    def _fetch(self, query: str, *args) -> List[BookmarkVar]:
-        self.cur.execute(query, args)
-        return [BookmarkVar(*x) for x in self.cur.fetchall()]
+    def _fetch(self, query: str, *args, lock: bool = True) -> List[BookmarkVar]:
+        if not lock:
+            self.cur.execute(query, args)
+            return [BookmarkVar(*x) for x in self.cur.fetchall()]
+        with self.lock:
+            return self._fetch(query, *args, lock=False)
```

Comment:
This change only affects _concurrent_ access (which would throw an error before the fix)